tools/estimate_cost.py

```
@@ -0,0 +1,308 @@
+    raise
+
+def encoding_for_model_key(model_key: str):
+    cfg = PRICING.get(model_key, {})
+    enc_name = cfg.get("tokenizer") or "o200k_base"
+    try:
+        return tiktoken.get_encoding(enc_name)
+    except Exception:
+        return tiktoken.get_encoding("o200k_base")
+
+JP_RX = re.compile(r"[぀-ヿ㐀-鿿々〆ヵヶ]")
+BR_RX = re.compile(r"
```

Ruff (0.12.2)

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected an expression or a ')'

---

SyntaxError: Expected an expression or a ')'

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected ')', found newline

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected ',', found name

---

SyntaxError: Expected ')', found NonLogicalNewline

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Expected 'else', found ':'

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: Expected ',', found ';'

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: missing closing quote in string literal

---

SyntaxError: Expected ')', found NonLogicalNewline

---

SyntaxError: Compound statements are not allowed on the same line as simple statements

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Compound statements are not allowed on the same line as simple statements

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Simple statements must be separated by newlines or semicolons

---

SyntaxError: Expected newline, found NonLogicalNewline

---

Multiple statements on one line (semicolon)

(E702)